gssapi/tests/test_high_level.py

```diff
@@ -380,18 +380,16 @@ def test_indicate_mechs(self):
     @ktu.gssapi_extension_test('rfc5801', 'RFC 5801: SASL Names')
     def test_sasl_properties(self):
         mechs = gssmechs.Mechanism.all_mechs()
-        encoding = gssutils._get_encoding()
         for mech in mechs:
             s = str(mech)
             s.shouldnt_be_empty()
             s.should_be_a(str)
-            s.should_be(mech._saslname.mech_name.decode(encoding))
 
-            mech.sasl_name.shouldnt_be_empty()
-            mech.sasl_name.should_be_a(six.text_type)
+            if mech.sasl_name:
```

Comment:
add a comment as to why this is conditional

Reply:
This should be resolved.